jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java

```diff
@@ -86,6 +86,12 @@ static boolean refKindIsConstructor(byte kind) {
 	static Object classData(Class<?> c) {
 		return JLA.classData(c);
 	}
+
+	private native static void checkClassBytesImpl(byte[] bytes);
```

Comment:
Now that there's no real code in the forwarder, would it make sense to just make checkClassBytes native?

Reply:
On second thought, how about moving the null and size = 0 check into the java wrapper code and remove it from the native code.

Reply:
Actually, since you're already fetching the length in the native (for more than just the 0 check), it may as well stay in the native.

Reply:
Changed checkClassBytes to native.